apps/src/componentLibrary/button/Button.tsx

```diff
@@ -1,36 +1,250 @@
-import React from 'react';
+import React, {memo, useMemo} from 'react';
 import classNames from 'classnames';
 
 import {ComponentSizeXSToL} from '@cdo/apps/componentLibrary/common/types';
+import FontAwesomeV6Icon, {
+  FontAwesomeV6IconProps,
+} from '@cdo/apps/componentLibrary/fontAwesomeV6Icon';
+
 import moduleStyles from './button.module.scss';
 
+type ButtonType =
+  | 'primary'
+  | 'secondary'
+  | 'tertiary'
+  | 'iconBorder'
+  | 'iconOnly';
+
+type ButtonColor = 'purple' | 'black' | 'white';
+
+export const buttonColors: {[key in ButtonColor]: ButtonColor} = {
+  purple: 'purple',
+  black: 'black',
+  white: 'white',
+};
+
 export interface ButtonProps {
```

Comment:
There are a lot of props here with a lot of underlying assumptions about which props are intended for which purpose. What do you think about exporting separate components, ex. `LinkButton`, `IconButton`, and `Button`/`TextButton` (for text + icons left/right) for the different variants, where we can ensure that the right props are being passed? Then we could turn this component into an internal-only component (`ButtonInternal` or `BaseButton` or maybe `CustomButton` to match the dropdown naming)

Reply:
Yeah, that's an option I think. Not sure though, since it might bring additional extra complexity to the code and will also require additional tests and stories.

The initial reason I decided to implement Button as is right now - following the skeleton of our current `Button` component, meaning transition to new `Button` component will be smother.

Also, with `Dropdown` the logic behind was that every dropdown had some differences visually and structurally, but here with `Button` all three types visually and structurally are almost identical.

Also, additional validation is provided here with help of `checkButtonPropsForErrors` function that can be extended as much as we want/need.

Finally, don't have a strong/leaning towards opinion right now, need to think what will be more convinient for future development and better consumer experience